### PR TITLE
Replace makefile.shade

### PR DESCRIPTION
**Pull request by natemcmaster**
Port sake targets to MSBuild.

Also, updated the README.md and added an issue template to make it clear that these tools are meant for our use, not general consumption.

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc @jbagga - BuildTools doesn't currently build with the latest version of korebuild. Can you take a look at these changes so we can get unblocked on building this repo?

**Comment by natemcmaster** [comment]
> The publish directory is missing from artifacts. Is it not required anymore?

That was an implementation detail of how we built these packages in Sake. We don't need it.

File: src/DependenciesPackager/DependenciesPackager.csproj [hunk 1]
```diff
@@ -18,4 +18,17 @@
...
   </ItemGroup>
 
+  <!-- packaging settings -->
+  <PropertyGroup>
+    <NoPackageAnalysis>true</NoPackageAnalysis>
+    <NuspecFile>$(MSBuildThisFileDirectory)$(MSBuildProjectName).nuspec</NuspecFile>
+  </PropertyGroup>
+
+  <Target Name="SetPackageDependencies" BeforeTargets="GenerateNuspec" DependsOnTargets="Publish">
+    <PropertyGroup>
```
Comment (javiercn) [inline src/DependenciesPackager/DependenciesPackager.csproj]: You should have a property named SetPackageDependenciesDependsOn and do this through <Target Name="SetPackageDependencies" DependsOnTargets="$(SetPackageDependenciesDependsOn)">

Mixing BeforeTargets and DependsOnTargets is kind of ugly and using properties with DependsOnTargets allows you to:
1) Allow third party scripts to plug in transparently into your build script by just defining <Property>$(Property);newvalue</Property>
2) Allows you to see all the dependency graph of tasks in one place. (the property group where all these things are defined.
3) You don't have to be hunting around to modify the dependency graph.

File: src/Microsoft.AspNetCore.BuildTools.ApiCheck/Microsoft.AspNetCore.BuildTools.ApiCheck.csproj [hunk 2]
```diff
@@ -21,4 +21,29 @@
...
+    <NoPackageAnalysis>true</NoPackageAnalysis>
+    <NuspecFile>$(MSBuildThisFileDirectory)$(MSBuildProjectName).nuspec</NuspecFile>
+    <IntermediatePackDir>$(MSBuildThisFileDirectory)bin\$(Configuration)\publish\</IntermediatePackDir>
+    <PublishDir>$(IntermediatePackDir)$(TargetFramework)\</PublishDir>
+    <NuspecBasePath>$(IntermediatePackDir)</NuspecBasePath>
+  </PropertyGroup>
+
+  <Target Name="PublishAll">
+    <ItemGroup>
+      <_TargetFrameworks Remove="@(_TargetFrameworks)"/>
```
Comment (javiercn) [inline src/Microsoft.AspNetCore.BuildTools.ApiCheck/Microsoft.AspNetCore.BuildTools.ApiCheck.csproj]: What is this exactly doing?

File: samples/NuGetPackageVerifier.Sample/NuGetPackageVerifier.Sample.proj [hunk 3]
```diff
@@ -1,7 +1,8 @@
-﻿<Project ToolsVersion="15.0">
+﻿<Project>
```
Comment (natemcmaster) [inline samples/NuGetPackageVerifier.Sample/NuGetPackageVerifier.Sample.proj]: Not required for this sample.

File: src/Microsoft.AspNetCore.BuildTools.ApiCheck/Microsoft.AspNetCore.BuildTools.ApiCheck.csproj (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.BuildTools.ApiCheck/Microsoft.AspNetCore.BuildTools.ApiCheck.csproj]: Clear the itemgroup and does a cross-targeting build. The "Publish" task requires you set a TargetFramework.

File: src/DependenciesPackager/DependenciesPackager.csproj (hunk 1 above)
Comment (natemcmaster) [inline src/DependenciesPackager/DependenciesPackager.csproj]: That may be true when developing MSBuild targets in an SDK, but seems like overkill for this one target in this one project.

Also, FWIW the SDK team is slowly trying to move off *DependsOn properties. These are the legacy of the first MSBuild which didn't have AfterTargets/BeforeTargets.

**Pull request merged**